datajunction-server/datajunction_server/api/materializations.py

```diff
@@ -316,51 +316,96 @@ async def list_node_materializations(
 
 @router.delete(
     "/nodes/{node_name}/materializations/",
-    response_model=List[MaterializationConfigInfoUnified],
+    response_model=None,
     name="Deactivate a Materialization for a Node",
 )
 async def deactivate_node_materializations(
     node_name: str,
     materialization_name: str,
+    node_version: str | None = None,
     *,
     session: AsyncSession = Depends(get_session),
     request: Request,
     query_service_client: QueryServiceClient = Depends(get_query_service_client),
     current_user: User = Depends(get_and_update_current_user),
     save_history: Callable = Depends(get_save_history),
-) -> List[MaterializationConfigInfoUnified]:
+) -> JSONResponse:
     """
     Deactivate the node materialization with the provided name.
     Also calls the query service to deactivate the associated scheduled jobs.
+
+    If node_version not provided, it will deactivate the materialization
+    for the current version of the node.
     """
     request_headers = dict(request.headers)
-    node = await Node.get_by_name(session, node_name)
+
+    # find the node revision to deactivate the materialization for
+    node_revision = None
+    if node_version:
+        node = await Node.get_by_name(
+            session,
+            node_name,
+            options=[
+                joinedload(Node.revisions).options(
```

Comment:
Can this part just ask the database to query for the single revision `node_version` that it needs? This setup with `joinedload(Node.revisions)` will end up loading all revisions for the node, which could be a lot.

Maybe something like:
```
stmt = (
    select(NodeRevision)
    .options(*NodeRevision.default_load_options())
    .where(NodeRevision.name == node_name, NodeRevision.version == node_version)
)
result = await session.execute(stmt)
node_revision = result.scalars().first()

if not node_revision:
    raise DJDoesNotExistException(...)
```

Reply:
Thanks! I'll try that.